### ISSUE TITLE
`no-setup-in-describe` triggers on both `describe.each()` and `it.each()`

### ISSUE THREAD

**Issue by maikeru** [lo1tuma/eslint-plugin-mocha#348]
Thanks for all the hard work on this incredibly useful plugin!

It looks like `describe.each()` and `it.each()` aren't handled and trigger the `no-setup-in-describe` rule when they shouldn't 
Plain `describe()` and `it()` behave as expected (i.e. don't trigger `no-setup-in-describe`. 

Some examples:
```js
import expect from 'expect';

// this triggers no-setup-in-describe
describe.each([['a'], ['b']])('letter %s', letter => {
  it('does something with the letter', () => {
    // meaningless test so that example is runnable
    expect(letter).toEqual(letter);
  });
});

describe('fruit', () => {
  // this also triggers no-setup-in-describe
  it.each([['apples'], ['oranges']])('eats %s', fruit => {
    expect(eats(fruit)).toBe(true);
  });
});
```

package.json dependencies:
```json
"eslint-plugin-mocha": "^10.3.0",
"expect": "29.6.1",
"mocha": "10.0.0",
```

Could it be a similar problem to this? https://github.com/lo1tuma/eslint-plugin-mocha/issues/249

**Comment by lo1tuma** [comment]
🤔  Mocha doesn’t have `describe.each()` or `it.each()`, so I think you need to specify `additionalCustomNames` so the rule knows about it and treats it as a describe or it block, see [here](https://github.com/lo1tuma/eslint-plugin-mocha?tab=readme-ov-file#plugin-settings).
